controlplane/kubeadm/internal/control_plane.go

```diff
@@ -238,24 +238,32 @@ func (c *ControlPlane) IsEtcdManaged() bool {
 	return c.KCP.Spec.KubeadmConfigSpec.ClusterConfiguration == nil || c.KCP.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd.External == nil
 }
 
-// UnhealthyMachines returns the list of control plane machines marked as unhealthy by MHC.
-func (c *ControlPlane) UnhealthyMachines() collections.Machines {
+// UnhealthyControlPlaneOrNodeMissingMachines returns all unhealthy control plane machines that
+// do not have a Kubernetes node, or have unhealthy control plane components.
+//
+// It differs from UnhealthyMachinesByHealthCheck which checks `MachineHealthCheck` conditions.
+func (c *ControlPlane) UnhealthyControlPlaneOrNodeMissingMachines(machines collections.Machines) collections.Machines {
```

Comment:
```suggestion
func (c *ControlPlane) UnhealthyMachinesByControlPlaneComponentsUnhealthyOrNodeMissing(machines collections.Machines) collections.Machines {
```
Should be more consistent with the other funcs this way

Reply:
Thanks @sbueringer for the review. I renamed the func to `UnhealthyMachinesWithMissingNodeOrUnhealthyControlPlaneComponents`, so as to be more consistent with the other funcs and also with the func `collections.HasMissingNodeOrUnhealthyControlPlaneComponents(c.IsEtcdManaged())` it calls.

Reply:
Sounds good, thx!